src/main/dialogs/captureToolDialog.js

```diff
@@ -0,0 +1,294 @@
+            args: ["{that}", "{flowManager}"]
+        }
+    },
+    listeners: {
+        onCreate: [
+            {
+                func: "{that}.show"
+            },
+            {
+                funcName: "gpii.app.captureTool.init",
+                args: ["{that}", "{flowManager}"]
+            }
+        ]
+    },
+    modelListeners: {
+        // This is a little flimsy. Ideally we'd like to send over
+        // one update rather than two. gpii-app seems to always have
+        // isKeyedIn updated before the pspChannel updates the preferences
+        // (which makes sense), so we listen for changes to preferences.
+        // Does it even matter though? If someones preferences changed, we
+        // would need to update the prefsset dropdown on page 4 too...
+        //
+        // So really there could be 2 situations. A change in key-in, which means
+        // we'd listen for isKeyedIn/keyedInUserToken and preferences. And second,
+        // the same user is keyed in, but their preferences changed.  Perhaps we
+        // should cache a copy of isKeyedIn/keyedInUserToken and set up 2
+        // listeners to handler this together.  Can you boil modelListeners/events?
+        // TODO
+        // This does actually cause an issue if you save a capture as a new prefset,
+        // then run the capture tool again it does not show up, but requires keying
+        // out and in again (this maybe be a different issue in using the channel).
+        // "isKeyedIn": {
+        "preferences": {
+            func: "{that}.updateRenderModel",
+            excludeSource: "init",
+            args: ["{pspChannel}.model.preferences", "{change}"]
+        }
+    }
+});
+
+/**
+ * This function will create a set of diagnostic log dumps and files for analyzing
+ * the output of capture on a machine. It will create a directory in the install
+ * directory with the current timestamp named `capture-diagnostics-TIMESTAMP` which
+ * will include the following files:
+ *
+ * + gpiiSettingsDir/
+ *   + capture-diagnostics-TIMESTAMP
+ *     - solutions.json
+ *     - installedSolutions.json
+ *     - settingsCapture.json
+ *
+ * @param {gpii.app.captureToolUtils} captureToolUtils = An instance of `gpii.app.captureToolUtils`
+ */
+gpii.app.captureTool.logCaptureDiagnostics = function (captureToolUtils) {
+    var settingsDirComponent = gpii.settingsDir();
+    var gpiiSettingsDir = settingsDirComponent.getGpiiSettingsDir();
+
+    var startupTime = Date.now();
+    var captureDirName = gpiiSettingsDir + "/capture-diagnostics-" + gpii.journal.formatTimestamp(startupTime);
+
+    fluid.log("Capture Diagnostics directory is: ", captureDirName);
+
+    var dailogPromise = electron.dialog.showMessageBox({
+        buttons: ["Cancel", "Run Capture"],
+        message: "Select 'Run Capture' to run the capture diagnostics and store them to disk.\n\n  After capture they will be stored in:\n" + captureDirName
+    });
+
+    if (dailogPromise === 1) {
```

Comment:
What's the significance of the number 1? How could a promise take on the value 1?

Reply:
Added a comment about how this is the return value from the electron promise implementation of their dialogs.